biojava-structure/src/main/java/org/biojava/nbio/structure/HetatomImpl.java

```diff
@@ -327,6 +329,71 @@ public boolean hasAminoAtoms(){
 
 	}
 
+	@Override
+	public boolean isPolymeric() {
+
+		ChemComp cc = getChemComp();
+
+		if ( cc == null)
+			return getType().equals(GroupType.AMINOACID) || getType().equals(GroupType.NUCLEOTIDE);
+
+		ResidueType rt = cc.getResidueType();
+
+		if ( rt.equals(ResidueType.nonPolymer))
+			return false;
+
+		PolymerType pt = rt.getPolymerType();
+
+		return pt.equals(PolymerType.peptide) ||
+				pt.equals(PolymerType.dna) ||
+				pt.equals(PolymerType.rna) ||
+				pt.equals(PolymerType.dnarna);
```

Comment:
In here we could use the `PolymerType.PROTEIN_ONLY` and `PolymerType.POLYNUCLEOTIDE_ONLY` predefined sets. This is for instance what we have in some methods in StructureTools:

```java
 PolymerType.PROTEIN_ONLY.contains(cc.getPolymerType())
	|| PolymerType.POLYNUCLEOTIDE_ONLY.contains(cc.getPolymerType()
```
Also in StructureTools and other places we check this too:
```java
ResidueType.lPeptideLinking.equals(cc.getResidueType())
```
Would that be useful here?